pkg/agent/plugin/workloadattestor/docker/retry.go

```diff
@@ -0,0 +1,58 @@
+package docker
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/spiffe/spire/test/clock"
+)
+
+const (
+	defaultNumRetries     = 3
+	defaultInitialBackoff = 100 * time.Millisecond
```

Comment:
We have similar logic in the kubernetes workload attestor as the workload can often times beat the k8s state update. Interestingly, this data comes from the docker event api, so we're probably actually working around the same thing.

The kubernetes workload attestor exposes some of this stuff as undocumented configurables. We recently found ourselves in a situation where we had to tweak them to deal with a demo that turned a bunch of stuff on at the same time. Might be worthwhile to similarly expose these, but this change is a strict improvement over the current posture so I'm happy to accept as-is.